### PR TITLE
fix: GitIgnoreManager to use exclusion rules from .git/info/exclude

### PR DESCRIPTION
**Pull request by nelsonni**
Adding support for exclusion rules from the `$GIT_DIR/info/exclude` file (per https://git-scm.com/docs/gitignore), based on the `TODO` tag in `GitIgnoreManager.js`:

https://github.com/isomorphic-git/isomorphic-git/blob/89c0da78d5ebf3c9f2754b3c8d557155dd70c8d7/src/managers/GitIgnoreManager.js#L7-L12

The contents of `$GIT_DIR/info/exclude` are in the same format as any `.gitignore` file, so reading and parsing the rules should be the same. Since the ignore manager from [`ignore`](https://www.npmjs.com/package/ignore) already handles removing comments and any other non-functional lines, we can simply add both rulesets into the ignore manager via [`ignore.add(patterns: Array<string | Ignore>)`](https://www.npmjs.com/package/ignore#addpatterns-arraystring--ignore-this).

### CODE REVIEW

File: src/managers/GitIgnoreManager.js [hunk 1]
```diff
@@ -6,15 +6,14 @@ import { join } from '../utils/join.js'
...
-// TODO: Implement .git/info/exclude
-
 export class GitIgnoreManager {
   static async isIgnored({ fs, dir, gitdir = join(dir, '.git'), filepath }) {
     // ALWAYS ignore ".git" folders.
     if (basename(filepath) === '.git') return true
     // '.' is not a valid gitignore entry, so '.' is never ignored
     if (filepath === '.') return false
+    // Load exclusion rules from the project exclude file (.git/info/exclude)
+    const excludes = await fs.read(join(gitdir, 'info', 'exclude'), 'utf8')
```
Comment (jcubic) [inline src/managers/GitIgnoreManager.js:16]: I would add check if the file exists first.

**Review by jcubic — CHANGES_REQUESTED** [review]
Thanks for the PR.

File: src/managers/GitIgnoreManager.js (hunk 1 above)
Comment (nelsonni) [reply src/managers/GitIgnoreManager.js]: Fixed in dcad82d.

**Comment by wmhilton** [comment]
This is fantastic!!! Thanks!!

**Pull request merged**